### ISSUE TITLE
Option to set the falloff in "Group Boosts"

### ISSUE THREAD

**Issue by smatthie** [mastodon/mastodon#18693]
### Pitch

The falloff for the "group boosts" feature is currently hard-coded to 40 but should be an option to be set by the user.

### Motivation

The setting "group boosts" hides multiple boosts, but after 40 other posts the next boost is shown in the timeline again.

For many users with many followers this can be reached in minutes so that they still see the same reboost every few minutes. Some users even never want to see reboosts, so they should have the option of increasing this value.

Also, I can't see any particular reason or motivation why it is set to 40 rather than some other value and generally I feel it's not good practice to have such values hard coded.

**Comment by smatthie** [comment]
See also https://github.com/mastodon/mastodon/issues/12604

**Comment by FediVideos** [comment]
Yes, this is a great idea.

Some people will see a boost again and again and again due to their browsing habits, it should be up to the user to decide how long boosts are grouped for.

I'd **strongly** back letting them set it by time rather than number of boosts, because this aligns more closely with what people expect, and avoids unexpected boost repetitions if scroll rate varies suddenly.

And as original poster says, some people never want to see the same boost twice. This ought to be an option too.

**Comment by FediVideos** [comment]
Also, the need to see re-boosts is much less nowadays thanks to the Explore feature. If someone is genuinely interested in highly boosted posts they can click on "Explore" rather than see the same boost over and over again in their main timeline.

**Comment by resuna** [comment]
New to Mastodon and my timeline is flooded in repeated boosts. I had no idea there was even a token effort at throttling... and it should throttle by reps not toots - only show me every 40th reboost of this post, not show me this boost every 40 toots.

**Comment by FediVideos** [comment]
This needs to be user (or at least admin) adjustable. Have had feedback from new people asking why they are seeing the same boosts again and again.

40 is way too low, and it's odd that any value is hardcoded considering how different instances and users' timelines can be.

**Comment by JorgeStolfi** [comment]
I don't want to EVER see the same toot twice in my timeline.  Why not suppress a boost if (1) I am following the sender of the original toot OR (2) the toot was already boosted by someone else who I follow?

**Comment by bdesham** [comment]
Just as an indication of how annoying people find this, the forthcoming [Ivory](https://tapbots.com/ivory/) client—which has been getting a tremendous amount of buzz—lists “suppress duplicate boosted posts” as one of the ten items on their roadmap.

**Comment by resuna** [comment]
I would much rather follow people who reboost interesting stuff a lot, because that gets me an interesting feed, but I can't do that because otherwise this bug floods me with tumbleweeds. I'm also not reboosting as much as I want to, because I don't want to flood people with rerererereboosts. This just creates an image of Mastodon as a ghost town.

**Comment by 31SFX4** [comment]
There are some good suggestions in this thread for further improvement, but I like to focus on my original request because it would be a trivial change to make: the variable already exists and has been tested and works etc. and it's just a matter of making it a variable that the user can set, rather than a constant.

This simple change should not require much work and testing. 

I agree that some other system (time limits etc) might be better in the end, but it's more complicated and therefore much less likely actually to be implemented, considering that there is a huge backlog of issues. So perhaps keep it simple for the moment?

**Comment by FediVideos** [comment]
Any progress on allowing people to increase the falloff number? This would be a really simple and useful change!

I've just seen more people complaining that they see too many boosts repeated. One person said just now they no longer boost anything because they don't want to contribute to the problem. This will harm the visibility of posts on the Fediverse, and it could be fixed by allowing users to change the falloff number.

**Comment by l-mb** [comment]
Another (additional?) option might be to not see boosted posts originally people you follow anyway.

Otherwise, yes, I'd like to see a 24h - 7d option for filtering duplicate boosts. Appreciated!

**Comment by tgarrison3** [comment]
> New to Mastodon and my timeline is flooded in repeated boosts. I had no idea there was even a token effort at throttling... and it should throttle by reps not toots - only show me every 40th reboost of this post, not show me this boost every 40 toots.
> 
> Edit: I just went through and unfollowed multiple people who primarily boost toots because otherwise every time I sat down at Mastodon I'd have hundreds of "new" toots most of which I'd already read.

You can also modify your follow of each user to not show their boosts, only their original content.

**Comment by babbage** [comment]
+1 for an option to not see a re-boosted post again until a certain time period has passed:
- an hour
- a day
- a week
- a month

As a user, I'd initially thought "forever" would be a good option here but that'd require tracking what had been boosted on to every timeline, forever, which seems untenable.

**Comment by HeckinTeagan** [comment]
voicing support for this idea.

**Comment by SylvKT** [comment]
I think it should still be possible to set it to 0, of course.

**Comment by doppelgrau** [comment]
I also think that it should be adjustable.
A given time limit would probably the best user experience (I'd set it probably to two or three days), but just allowing the user to modify the existing "number of posts searched" seems like a low effort (potentially temporary) solution.

**Comment by velw** [comment]
Just adding my +1 to this - have noticed the same issue myself. Would be great to see this configurable by time period rather than number of posts. The suggestions here look great.

**Comment by Dewines** [comment]
I agree that this should be a longer period and preferably something that can be modified by each user to suit their own preference. I have had to turn off boosts from quite a few of the people I follow because my timeline had too many repeated posts.

**Comment by tv21** [comment]
I agree with this, I am so sick of seeing the same posts over and over.  And also, I would like it to NEVER re-show anything I have boosted.  I've obviously already seen anything I have boosted, so why am I being shown it again?  The fact that you are re-shown anything you boost seems like a disincentive to boosting anything!

**Comment by valiant8086** [comment]
Yes, I never want to have the same boost twice.

**Comment by kaiserkiwi** [comment]
I would love an option like: Show me boosts again every nth time it's boosted. If the post gets boosted once and after 40 posts get boosted again nothing has really changed for this post. But if the post get boosted 10 times it seems to be a bit more interesting or important. So it's more interesting to see it again because the conversation under this post could be much more rich.

**Comment by q-rai** [comment]
I would love any form of limiting seeing boosts repeatedly. My preference would be age-based (if it's been a week/a month/... since I last saw it), but a number of toots since I last saw it would be fine, too. (Or never)

**Comment by FediVideos** [comment]
Any more progress on this? Had more feedback from users saying how frustrating repeated boosts are. The 40 number is clearly way too low for current users.

**Comment by davidak** [comment]
I think a sane default would be to not show the same post again on the same day (or for 24 hours?) with an option to never show a post again. But then an instance has to track which posts i have _seen_, which might not be technically feasible or wanted for privacy reasons.

**Comment by gnate-m** [comment]
I agree, whichever option has the lowest barrier to implementation would be an improvement, even if it's just per instance. Of course, I'd prefer per user, since frequency of timeline visits will vary greatly, but only if it's a trivial amount of additional work.


> There are some good suggestions in this thread for further improvement, but I like to focus on my original request because it would be a trivial change to make: the variable already exists and has been tested and works etc. and it's just a matter of making it a variable that the user can set, rather than a constant.

**Comment by resuna** [comment]
So long as I can set it arbitrarily high.

**Comment by emgrasso** [comment]
I use the web gui, which flags posts that I myself have favorited or boosted by changing the color of the icons. Therefore the GUI absolutely knows that I have seen those posts already, because I have interacted with them. I do not need to see in my Home stream any post that I myself have favorited or boosted (though it would be nice to have a show-all toggle...).  It would make sense to include them in expanded threads to avoid disrupting the conversation.

**Comment by l-mb** [comment]
Any progress on this one? Can we at least get an indiction whether it will be considered or not?

The more my fediverse connections grow, the more relevant this becomes. Currently, it is approaching unusability.

**Comment by tv21** [comment]
If people start preferring Bluesky over Mastodon, the fact that this hasn't been addressed will be one of the major reasons why!

**Comment by FediVideos** [comment]
Still seeing people asking on Feditips about this, they don't even realise there is a Group Boosts feature because they're seeing the same boosts so often! The 40 number is clearly way too low for the modern Fediverse.

**Comment by akkana** [comment]
I'd prefer being able to control visibility by time rather than number of toots, but either one would be helpful. Reboosts once or twice a day is helpful in case I missed it the first time, but seeing the same toot every hour or two, not so much.

**Comment by VulcanTourist** [comment]
Please PLEASE implement this proposed solution.  The issue being problematic for no less than two years is long enough, don't you think?

**Comment by LasseGismo** [comment]
Waiting for that feature since I'm on Masto - would love to get it.

**Comment by lrreynolds** [comment]
I agree this is one of the few really annoying things about browsing the mastodon timeline. Not sure exactly what the best user experience would be but allowing some user driven setting for this filter would probably take away much of the frustration  of seeing the same post boosted dozens of times.

**Comment by Eli-the-Bearded** [comment]
Why is this still happening?

I just had my own toot from yesterday cross my path twice today because of boosts. Guess what? I haven't forgotten the things I looked at yesterday, and I certainly haven't forgotten the things I sent out yesterday. I want it suppressed from view so I can see the new stuff.

**Comment by cymplecy** [comment]
Just found out why I'm seeing boosted post multiple times and would like facility to adjust this

**Comment by Cassolotl** [comment]
I think if I had the power to control this I wouldn't have an upper limit of boost grouping, and I wouldn't have the system track which posts I'd seen. I would just set it to only show me the same boost once every 24 hours.

**Comment by Nicthurne** [comment]
Looks like the 40 threshold will be raised to 80 as per this pr?
 https://github.com/mastodon/mastodon/pull/39784
